llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -4719,6 +4733,14 @@ VectorizationFactor LoopVectorizationPlanner::selectEpilogueVectorizationFactor(
                "Trip count SCEV must be computable");
         RemainingIterations = SE.getURemExpr(
             TC, SE.getConstant(TCType, MainLoopVF.getKnownMinValue() * IC));
+        const APInt MaxRemainingIterations =
+            SE.getUnsignedRangeMax(RemainingIterations);
+        // Guard against huge trip counts.
+        if (MaxRemainingIterations.getActiveBits() <= 32) {
```

Comment:
Hm, I would assume the max would always be `MainLoopVF.getKnownMinValue() * IC - 1` or less, if the trip count is known to be constant? Curios why this guard is needed?

Reply:
Right, it shouldn't be, I was overly defensive there.

Reply:
Added `MainLoopVF.getKnownMinValue() * IC - 1` as an upper bound, i.e., only use `RemainingIterations` if it's known to be smaller